scripts/get-nightly-build-pass-rate.ts

```diff
@@ -0,0 +1,108 @@
+import puppeteer from 'puppeteer';
+import dotenv from 'dotenv';
+dotenv.config();
+
+const loginAndAnalyzeBuilds = async (
+  latestBuildNumber: number,
+  numberOfDays: number,
+  username: string,
+  token: string
+) => {
+  const loginUrl = process.env.JENKINS_LOGIN_URL;
+  const baseApiUrl = process.env.JENKINS_BASE_API_URL;
+  const browser = await puppeteer.launch({ headless: true });
+  const page = await browser.newPage();
+
+  if (!loginUrl || !baseApiUrl) {
+    console.error('Missing environment variables. Please check your .env file.');
+    return;
+  }
+
+  try {
+    console.log('Logging in to Jenkins...');
+    await page.goto(loginUrl, { waitUntil: 'networkidle2' });
+    await page.type('#j_username', username);
+    await page.type('input[name="j_password"]', token);
+    await Promise.all([
+      page.click('button[type="submit"]'),
+      page.waitForNavigation({ waitUntil: 'networkidle2' }),
+    ]);
+    console.log('Login successful.');
+
+    const startBuildNumber = latestBuildNumber - numberOfDays + 1;
+    console.log(`Analyzing builds from ${startBuildNumber} to ${latestBuildNumber}...`);
+
+    const failedBuilds: number[] = [];
+    let passCount = 0;
+
+    for (let buildNumber = startBuildNumber; buildNumber <= latestBuildNumber; buildNumber++) {
+      console.log(`Fetching data for build #${buildNumber}...`);
+      const url = `${baseApiUrl}/${buildNumber}/api/json`;
+      try {
+        const response = await page.goto(url, { waitUntil: 'networkidle0', timeout: 10000 });
+
+        if (!response) {
+          console.log(`Build #${buildNumber}: No response from server.`);
+          continue;
+        }
+
+        const responseBody = await response.text();
```

Comment:
Could you call `response.json()` here and remove the later call to JSON.parse? Maybe not since Puppeteer is returning HTML?

Reply:
So, I didn't think it was guaranteed that the response would always be JSON. Since we're retrieving page content, it might return HTML or plain text, so I felt using response.text() and handling JSON parsing in a try/catch block was a safer approach for now. We can revisit and potentially refactor this once we move away from Puppeteer.